be/src/olap/olap_table.cpp

```diff
@@ -1867,8 +1867,19 @@ void OLAPTable::_list_files_with_suffix(const string& file_suffix, set<string>*
     }
 }
 
-bool OLAPTable::has_version(const Version& version) const {
-    return (_data_sources.find(version) != _data_sources.end());
+bool OLAPTable::has_segment_group(const Version& version, const SegmentGroup* new_segment_group) const {
+    if (_data_sources.find(version) == _data_sources.end()) {
```

Comment:
```suggestion
    auto it = _data_sources.find(version);
    if (it == _data_sources.end()) {
```
and you can avoid seek two times